src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/StringSyntaxAttribute.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Diagnostics.CodeAnalysis
+{
+    /// <summary>
+    /// <see cref="StringSyntaxAttribute"/> provides an interface to tell your IDE how a string is highlighted.
+    /// </summary>
+    /// <example>
+    ///     <code>
+    /// using System.Diagnostics.CodeAnalysis;
+    ///
+    /// public void MyRegexMethod([StringSyntax(StringSyntaxAttribute.Regex)] string myRegex) {
+    ///     // ...
+    /// }
+    ///     </code>
+    /// </example>
+    [AttributeUsage(AttributeTargets.Parameter |
+                    AttributeTargets.Field |
+                    AttributeTargets.Property)]
+    public sealed class StringSyntaxAttribute : Attribute
+    {
+        /// <summary>
+        /// Initializes <see cref="StringSyntaxAttribute"/> with <paramref name="syntax"/> in <see cref="Syntax"/>.
+        /// </summary>
+        /// <param name="syntax">The syntax identifier (see constant strings in <see cref="StringSyntaxAttribute"/>)</param>
+        public StringSyntaxAttribute(string syntax)
+        {
+            Syntax = syntax;
```

Comment:
looking at our other attributes with constructors taking strings, there seems to be a predominant pattern of not validating it for null/empty. so this is fine.

Reply:
also the compiler will validate for null.